airflow/api_fastapi/core_api/routes/public/task_instances.py

```diff
@@ -671,6 +673,11 @@ def patch_task_instance(
     fields_to_update = body.model_fields_set
     if update_mask:
         fields_to_update = fields_to_update.intersection(update_mask)
+    else:
+        try:
+            PatchTaskInstanceBody.model_validate(body)
+        except ValidationError as e:
+            raise RequestValidationError(errors=e.errors())
```

Comment:
Task instance is not doing exacly the same as the rest I believe.

We are not replacing `data` by the validated entity, therefore we are still using only `fields_to_update = body.model_fields_set` even when the `update_mask` is not provided.

Because for other endpoints we do:
```
data = patch_body.model_dump(exclude=non_update_fields, by_alias=True, exclude_none=True)
```

I think we should do the same.

Reply:
exclude_none should not be used right?

You mentioned if the field in None then also we should update it.
In both

>  cases I think we should only use set values. Things that the user explicitely specified in the payload. (independently if those are None or not None). You can take a look at patch_task_instance that leverages model_fields_set to only retrive explicitely set values.